mne/decoding/transformer.py

```diff
@@ -206,43 +207,68 @@ def transform(self, epochs_data, y=None):
         -------
         X : array, shape (n_epochs, n_channels * n_times)
             The data concatenated over channels
+        y : array, shape(n_epochs,)
+            The label for each epoch.
         """
-        if not isinstance(epochs_data, np.ndarray):
-            raise ValueError("epochs_data should be of type ndarray (got %s)."
-                             % type(epochs_data))
-
-        epochs_data = np.atleast_3d(epochs_data)
+        if not isinstance(epochs, _BaseEpochs):
+            raise ValueError("epochs should be an instance of Epochs got %s"
+                             " instead" % type(epochs))
 
-        n_epochs, n_channels, n_times = epochs_data.shape
-        X = epochs_data.reshape(n_epochs, n_channels * n_times)
+        n_epochs, n_channels, n_times = epochs._data.shape
+        X = epochs._data.reshape(n_epochs, n_channels * n_times)
         # save attributes for inverse_transform
-        self.n_epochs = n_epochs
-        self.n_channels = n_channels
-        self.n_times = n_times
+        self.n_epochs = len(X)
+        self.info = epochs.info
 
-        return X
+        if (y is None):
+            y = epochs.events[:, 2]
+        return X, y
 
-    def inverse_transform(self, X, y=None):
-        """For each epoch, reshape a feature vector into the original data shape
+    def fit_transform(self, epochs, y=None):
+        """First fit the data, then transform it
 
         Parameters
         ----------
+        epochs : instance of Epochs
+            The data.
+        y : None, array of shape(n_epochs,)
+            Label of epochs. If None, events are used.
+
+        Returns
+        -------
         X : array, shape (n_epochs, n_channels * n_times)
-            The feature vector concatenated over channels
-        y : None | array, shape (n_epochs,)
+            The data concatenated over channels
+        y : array, shape(n_epochs,)
```

Comment:
I agree with @kingjr if you stick to current sklearn API transform only returns X.